Sources/Commands/SwiftPackageTool.swift

```diff
@@ -687,6 +728,236 @@ extension SwiftPackageTool {
     }
 }
 
+#if BUILD_PACKAGE_SYNTAX
+extension SwiftPackageTool {
+    struct AddDependency: SwiftCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Add a dependency to the current package.")
+
+        @OptionGroup()
+        var swiftOptions: SwiftToolOptions
+
+        @Argument(help: "The URL of a remote package, or the path to a local package")
+        var dependencyURL: String
+
+        @Option(help: "Specifies an exact package version requirement")
+        var exact: Version?
+
+        @Option(help: "Specifies a package revision requirement")
+        var revision: String?
+
+        @Option(help: "Specifies a package branch requirement")
+        var branch: String?
+
+        @Option(help: "Specifies a package version requirement from the specified version up to the next major version")
+        var from: Version?
+
+        @Option(help: "Specifies a package version requirement from the specified version up to the next minor version")
+        var upToNextMinorFrom: Version?
+
+        func run(_ swiftTool: SwiftTool) throws {
+            var requirements: [PackageDependencyRequirement] = []
+            if let exactVersion = exact {
+                requirements.append(.exact(exactVersion.description))
+            }
+            if let revision = revision {
+                requirements.append(.revision(revision))
+            }
+            if let branch = branch {
+                requirements.append(.branch(branch))
+            }
+            if let version = from {
+                requirements.append(.upToNextMajor(version.description))
+            }
+            if let version = upToNextMinorFrom {
+                requirements.append(.upToNextMinor(version.description))
+            }
+
+            guard requirements.count <= 1 else {
+                swiftTool.diagnostics.emit(.error("only one requirement is allowed when specifiying a dependency"))
+                throw ExitCode.failure
+            }
+
+            let requirement = requirements.first
+
+            let packageRoot = try swiftTool.getPackageRoot()
+            let editor = try PackageEditor(manifestPath: packageRoot.appending(component: Manifest.filename),
```

Comment:
For now, I've switched this and the other two tools to using `Manifest.path(atPackagePath:fileSystem)`, so that the manifest we try to edit is the same one the current version of SwiftPM would use to build the package. I don't think that's too surprising, but we still might want to revisit it sometime.